src/audio_unit/looper.rs

```diff
@@ -88,24 +110,72 @@ impl Looper {
                 }
             }
             Playing { position, total } => {
+                self.process_samples_playing(position, total, input, output);
+            }
+            PlayingAwaitingOverdub { position, total } => {
+                self.process_samples_playing(position, total, input, output);
+            }
```

Comment:
I don't remember the syntax, but there is some way of having multiple cases which have the same body done in a single match.

Reply:
Conceptually what seems tricky about that is that all of the cases would have to put the same variables (I guess with the same types?) into scope in order for the shared body to make sense (eg here `position` and `total`)

Reply:
Right, it only allows it in the case that the variable types all line up:

```rust
            Playing { position, total } | PlayingAwaitingOverdub { position, total } => {
                self.process_samples_playing(position, total, input, output);
            }
```